presto-main/src/test/java/io/prestosql/server/TestDynamicFilterService.java

```diff
@@ -0,0 +1,275 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.stats.Distribution;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import io.prestosql.cost.StatsAndCosts;
+import io.prestosql.execution.StageId;
+import io.prestosql.execution.StageInfo;
+import io.prestosql.execution.StageState;
+import io.prestosql.execution.StageStats;
+import io.prestosql.execution.TaskId;
+import io.prestosql.execution.TaskInfo;
+import io.prestosql.execution.TaskManagerConfig;
+import io.prestosql.execution.TaskStatus;
+import io.prestosql.operator.TaskStats;
+import io.prestosql.spi.QueryId;
+import io.prestosql.spi.eventlistener.StageGcStatistics;
+import io.prestosql.spi.predicate.Domain;
+import io.prestosql.spi.type.IntegerType;
+import io.prestosql.sql.planner.Partitioning;
+import io.prestosql.sql.planner.PartitioningScheme;
+import io.prestosql.sql.planner.PlanFragment;
+import io.prestosql.sql.planner.Symbol;
+import io.prestosql.sql.planner.plan.JoinNode;
+import io.prestosql.sql.planner.plan.PlanFragmentId;
+import io.prestosql.sql.planner.plan.PlanNodeId;
+import io.prestosql.sql.planner.plan.TableScanNode;
+import io.prestosql.testing.TestingMetadata;
+import org.joda.time.DateTime;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.units.DataSize.Unit.BYTE;
+import static io.prestosql.execution.TaskInfo.createInitialTask;
+import static io.prestosql.operator.StageExecutionDescriptor.ungroupedExecution;
+import static io.prestosql.spi.type.VarcharType.VARCHAR;
+import static io.prestosql.sql.planner.SystemPartitioningHandle.SINGLE_DISTRIBUTION;
+import static io.prestosql.sql.planner.SystemPartitioningHandle.SOURCE_DISTRIBUTION;
+import static io.prestosql.sql.planner.plan.JoinNode.Type.INNER;
+import static io.prestosql.testing.TestingHandles.TEST_TABLE_HANDLE;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestDynamicFilterService
+{
+    @Test
+    public void testDFSummaryCompletion()
```

Comment:
We should also test result of:
```
createDynamicFilterSupplier
```
and use more than one `filterId` in test.

Reply:
Added `testDynamicFilterSupplier` which tests this